prediction_market_agent/utils.py

```diff
@@ -96,3 +97,25 @@ def patch_sqlite3() -> None:
         sys.modules["sqlite3"] = sys.modules.pop("pysqlite3")
     except ImportError:
         logger.warning("pysqlite3-binary not found, using sqlite3 instead.")
+
+
+def streamlit_asyncio_event_loop_hack() -> asyncio.AbstractEventLoop:
+    """
+    This function is a hack to make Streamlit work with asyncio event loop.
+    See https://github.com/streamlit/streamlit/issues/744
+    """
+
+    def get_or_create_eventloop() -> asyncio.AbstractEventLoop:
+        try:
+            return asyncio.get_event_loop()
+        except RuntimeError as ex:
+            if "There is no current event loop in thread" in str(ex):
+                loop = asyncio.new_event_loop()
+                asyncio.set_event_loop(loop)
+                return asyncio.get_event_loop()
+            else:
+                raise ex
+
+    loop = get_or_create_eventloop()
```

Comment:
Seems like this line of code is never achieved.

try will return, `if` in the except block will return and `else` in the except block will raise.

Reply:
Not sure I understand. Here it's calling the inline function above, so this is the first line that runs

Reply:
Ah, I didn't notice that that try-except code is inside of another function! 😄 🤦